### DISCUSSION TITLE
vercel dev shuts down with 'Error: socket hang up'

### DISCUSSION THREAD

**Discussion by mmgj** [vercel/vercel#6314] [Help] [answered]
I'm developing a Gatsby project and have so far been using the default `gatsby develop` with no issues. When I try using `vercel dev` instead in order to work on serverless functions the command crashes after a few seconds with the following error:

`Error: socket hang up
    at connResetException (node:internal/errors:683:14)
    at Socket.socketOnEnd (node:_http_client:471:23)
    at Socket.emit (node:events:377:35)
    at Socket.emit (node:domain:470:12)
    at endReadableNT (node:internal/streams/readable:1312:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)`
    
Things I've tried: 
* node versions 14, 15, 16 (using nvm and trashing node_modules and reinstalling for each switch)
* Changing the --listen port 
* Logging out and back in

The CLI _will_ let me deploy my project, and my serverless function seems to be running fine when deployed, but I kinda need to be able to work with serverless functions locally. 

Please let me know if there's additional info I should include in order to help you help me.

Regards,
-m

**Comment by mmgj** [comment, marked as answer]
Close this. It somehow worked out.

**Reply by bryce-hunt** [reply]
Running into this issue as well. What did you end up doing to resolve this?

**Reply by gvasilakiss** [reply]
Same here.

**Reply by wokejacqueline** [reply]
running into this issue as well. `npm run clean && vercel dev` seems to resolve locally. (clears the cache then runs vercel dev sever)

**Reply by mralbertchen** [reply]
That didn't work for me

**Reply by tsar-boomba** [reply]
For me all I had to do was switch the port I was running on from 3000 => 3001. Using ```vercel dev --listen 3001```.

**Reply by wokejacqueline** [reply]
**update on my previous comment ^** this still happens to me running `vercel dev` (vercel@23.1.2) for a gatsby `4.7.1` project. i noticed that if you close the browser tab while restarting the server, it starts working reliably.  i am suspicious that this process is connected to the browser tab somehow.

steps:
* close any / all browser tabs running your local dev server
* run`vercel dev`
* should work

if it doesn't work, you can also try running `killall node` before doing `vercel dev`.  closing the browser tab & restarting the server always worked for me, whereas changing the port number / clearing the cache only worked sometimes, and might just be coincidental

**Reply by beadlespouse** [reply]
Closing the browser tab seems to work for me. I guess that's why switching port works too.

**Reply by ZeldOcarina** [reply]
Yeah closing the tab works but it's still a bug and should be fixed!
